DarwinWorld/src/main/java/agh/ics/oop/model/utils/Genotype.java

```diff
@@ -41,20 +41,25 @@ public int getCurrentGene() {
     public List<Integer> getGenotypeList() {
         return genotype;
     }
-    private List<Integer> generateNewGenotype(Animal parent1, Animal parent2) {
-        int totalEnergy = parent1.getEnergy() + parent2.getEnergy();
-        List<Integer> parent1GenotypeList = parent1.getGenotype().getGenotypeList();
-        List<Integer> parent2GenotypeList = parent2.getGenotype().getGenotypeList();
+    private List<Integer> generateNewGenotype(Animal strongerParent, Animal weakerParent) {
+        int totalEnergy = strongerParent.getEnergy() + weakerParent.getEnergy();
+        if(AnimalComparator.compare(strongerParent, weakerParent) < 1) {
```

Comment:
czemu sprawdzamy czy na pewno stronger jest silniejszy skoro o to dba funkcja findParents()?

Reply:
Dodałem takiego checka bo w innym miejscu moglibyśmy też korzystać z tej funkcji i łatwo moglibyśmy podac w złej kolejności rodziców. Funkcja teoretycznie wymaga podania 2 zwierzątek i z zewnątrz nie widać wymogu w jakiej kolejności :/